### DISCUSSION TITLE
Running Textual in a browser?

### DISCUSSION THREAD

**Discussion by tedmiston** [Textualize/textual#2872] [Q&A] [answered]
I am curious to inquire if any work or experimentation has been done to run Textual in a browser?

I'm thinking about this as an alternative to building a GUI web app frontend — essentially a browser UI which acts like a very basic "mini pseudoterminal" to render the Textual TUI for an app.

I perused through the examples but didn't see any along the lines of this concept.

I also found a PDF linked from the docs that mentions a possibly forthcoming ["Textual Cloud Service" (PDF)][1] which sounds like it may be along these same lines (also #2624).  I did not see this service referenced on [the roadmap][2].

[1]: https://www.textualize.io/files/Textual%20Cloud%20Service%20-%20Google%20Docs.pdf

[2]: https://textual.textualize.io/roadmap/

There are several popular modern efforts to run or transpile Python code for a browser which may potentially be applicable here, such as PyScript, Brython, or Transcrypt.

**Comment by TomJGooding** [comment, marked as answer]
You might have missed this recent discussion:

- https://github.com/Textualize/textual/discussions/2765

**Comment by willmcgugan** [comment]
It is very much a work in progress. Expect an announcement in a few weeks. Join the Discord server if you would like to beta test...
